src/gil.rs

```diff
@@ -770,29 +760,26 @@ mod tests {
 
     #[cfg(not(target_arch = "wasm32"))]
     struct Event {
-        set: Mutex<bool>,
-        wait: Condvar,
+        set: sync::Mutex<bool>,
+        wait: sync::Condvar,
     }
 
     #[cfg(not(target_arch = "wasm32"))]
     impl Event {
         const fn new() -> Self {
             Self {
-                set: const_mutex(false),
-                wait: Condvar::new(),
+                set: sync::Mutex::new(false),
+                wait: sync::Condvar::new(),
             }
         }
 
         fn set(&self) {
-            *self.set.lock() = true;
+            *self.set.lock().unwrap() = true;
             self.wait.notify_all();
         }
 
         fn wait(&self) {
-            let mut set = self.set.lock();
-            while !*set {
-                self.wait.wait(&mut set);
-            }
+            drop(self.wait.wait_while(self.set.lock().unwrap(), |s| !*s));
```

Comment:
Why `drop` instead of `unwarp` the poisoning error here?

Reply:
It needed to be `drop(unwrap())`, but good catch.